packages/bitcore-wallet-client/src/lib/api.ts

```diff
@@ -643,18 +643,25 @@ export class API extends EventEmitter {
   }
 
   _addSignaturesToBitcoreTx(txp, t, signatures, xpub) {
-    switch (txp.coin) {
-      case 'eth':
-        const raw = CWC.Transactions.applySignature({
-          chain: 'ETH',
-          tx: t.uncheckedSerialize(),
-          signature: signatures[0],
-        });
-        t.uncheckedSerialize = () => raw ;
-        t.serialize = () => raw ;
+    const chain = Utils.getChain(txp.coin);
+    switch (chain) {
+      case 'ETH':
+        const unsignedTxs = t.uncheckedSerialize();
+        const iterator = new Array(...signatures).entries();
```

Comment:
Might want a better name than iterator, also may not need Array(...signatures).

A traditional `for(let index = 0; index < signatures.length; index++)` may be more clear

Reply:
changed all iterators to traditional for loops